benchmarks/commonMain/src/kotlinx/io/BytesInputReadBenchmark.kt

```diff
@@ -0,0 +1,67 @@
+package kotlinx.io
+
+import kotlinx.benchmark.*
+import kotlinx.io.bytes.*
+
+/**
+ * Benchmark with prebuilt ByteInput
+ */
+@State(Scope.Benchmark)
+@BenchmarkMode(Mode.Throughput)
+@OutputTimeUnit(BenchmarkTimeUnit.MICROSECONDS)
+class BytesInputReadBenchmark {
+    lateinit var input: Input
+
+    @Param("2097152")
+    open var size: Int = 42
+
+    @Setup
+    fun setup() {
```

Comment:
Have you managed to find real-world benchmarks previous IO was optimized against? Maybe ask @cy6erGn0m while we are here?

Reply:
They are in a separate branch.